### PR TITLE
Use predictable collections in classes that participate in planning

### PR DESCRIPTION
**Pull request by sopel39**

### CODE REVIEW

File: presto-spi/src/main/java/io/prestosql/spi/predicate/EquatableValueSet.java [hunk 1]
```diff
@@ -63,7 +64,7 @@ public EquatableValueSet(
         }
         this.type = type;
         this.whiteList = whiteList;
-        this.entries = Collections.unmodifiableSet(new HashSet<>(entries));
+        this.entries = Collections.unmodifiableSet(new LinkedHashSet<>(entries));
```
Comment (findepi) [inline presto-spi/src/main/java/io/prestosql/spi/predicate/EquatableValueSet.java]: while here, static import

**Comment by sopel39** [comment]
ac

File: presto-spi/src/main/java/io/prestosql/spi/connector/GroupingProperty.java [hunk 2]
```diff
@@ -37,7 +38,7 @@ public GroupingProperty(@JsonProperty("columns") Collection<E> columns)
     {
         requireNonNull(columns, "columns is null");
 
-        this.columns = Collections.unmodifiableSet(new HashSet<>(columns));
+        this.columns = Collections.unmodifiableSet(new LinkedHashSet<>(columns));
```
Comment (findepi) [inline presto-spi/src/main/java/io/prestosql/spi/connector/GroupingProperty.java]: static import

File: presto-spi/src/main/java/io/prestosql/spi/predicate/TupleDomain.java [hunk 3]
```diff
@@ -419,4 +420,9 @@ public Domain getDomain()
             return domain;
         }
     }
+
+    private static <T, K, U> Collector<T, ?, Map<K, U>> toLinkedMap(Function<? super T, ? extends K> keyMapper, Function<? super T, ? extends U> valueMapper)
+    {
+        return toMap(keyMapper, valueMapper, (u, v) -> { throw new IllegalStateException(String.format("Duplicate key %s", u)); }, LinkedHashMap::new);
```
Comment (findepi) [inline presto-spi/src/main/java/io/prestosql/spi/predicate/TupleDomain.java]: put each arg on separate line

File: presto-spi/src/main/java/io/prestosql/spi/predicate/EquatableValueSet.java [hunk 4]
```diff
@@ -246,20 +248,20 @@ public String toString(ConnectorSession session)
...
-                .collect(toSet());
+                .collect(toCollection(LinkedHashSet::new));
     }
 
     private static <T> Set<T> subtract(Set<T> set1, Set<T> set2)
     {
         return set1.stream()
                 .filter(value -> !set2.contains(value))
-                .collect(toSet());
+                .collect(toCollection(LinkedHashSet::new));
```
Comment (findepi) [inline presto-spi/src/main/java/io/prestosql/spi/predicate/EquatableValueSet.java]: maybe capture this as `toLinkedSet` helper function

**Comment by findepi** [comment]
Travis failure looks related
```
[ERROR] testMultipleColumnsMultipleRanges(io.prestosql.plugin.raptor.legacy.metadata.TestShardPredicate)  Time elapsed: 0.017 s  <<< FAILURE!
java.lang.AssertionError: expected [(((c1_max >= ? OR c1_max IS NULL) AND (c1_min <= ? OR c1_min IS NULL)) OR ((c1_max >= ? OR c1_max IS NULL) AND (c1_min <= ? OR c1_min IS NULL))) AND (((bucket_number >= ? OR bucket_number IS NULL) AND (bucket_number <= ? OR bucket_number IS NULL)) OR ((bucket_number >= ? OR bucket_number IS NULL) AND (bucket_number <= ? OR bucket_number IS NULL)))] but found [(((bucket_number >= ? OR bucket_number IS NULL) AND (bucket_number <= ? OR bucket_number IS NULL)) OR ((bucket_number >= ? OR bucket_number IS NULL) AND (bucket_number <= ? OR bucket_number IS NULL))) AND (((c1_max >= ? OR c1_max IS NULL) AND (c1_min <= ? OR c1_min IS NULL)) OR ((c1_max >= ? OR c1_max IS NULL) AND (c1_min <= ? OR c1_min IS NULL)))]
	at org.testng.Assert.fail(Assert.java:94)
	at org.testng.Assert.failNotEquals(Assert.java:513)
	at org.testng.Assert.assertEqualsImpl(Assert.java:135)
	at org.testng.Assert.assertEquals(Assert.java:116)
	at org.testng.Assert.assertEquals(Assert.java:190)
	at org.testng.Assert.assertEquals(Assert.java:200)
	at io.prestosql.plugin.raptor.legacy.metadata.TestShardPredicate.testMultipleColumnsMultipleRanges(TestShardPredicate.java:133)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Pull request merged**